c2corg_ui/static/js/search.js

```diff
@@ -163,44 +172,58 @@ app.SearchController = function($scope, $compile, $attrs, apiUrl, gettextCatalog
 
 /**
  * @param {string} type The document type.
- * @param {string} title The category title.
  * @return {TypeaheadDataset} A data set.
  * @private
  */
-app.SearchController.prototype.createDataset_ = function(type, title) {
+app.SearchController.prototype.createDataset_ = function(type) {
   var bloodhoundEngine = this.createAndInitBloodhound_(type);
   return /** @type {TypeaheadDataset} */({
+    contents : type,
     source: bloodhoundEngine.ttAdapter(),
     display: function(doc) {
       return doc.label;
     },
     limit: 20,
     templates: {
       header: (function() {
-        return '<div class="header ' + title + '">' +
-            this.gettextCatalog_.getString(title) + '</div>';
+        return '<div class="header" dataset="' + type + '">' + this.gettextCatalog_.getString(type) + '</div>';
       }).bind(this),
+      footer: function() {
+        return '<p class="suggestion-more"><a href="/' + type + '" class="green-text">+ see more results</a></p>';
+      },
       suggestion: function(doc) {
-        return '<p>' + doc.label + '</p>';
-      }
-      // Note: The templates for `notFound` and `pending` are not set, because
-      // they would be shown for each dataset.
+        var document = {
+          'query' : doc['_query'],
+          'elevation' : doc['elevation'] || doc['elevation_max'],
+          'waypoint_type' : doc['waypoint_type'],
+          'activities' : doc['activities'],
+          'date_end' : doc['date_end'],
+          'doctype' : doc['documentType']
+        };
+        return this.compile_('<app-suggestion label="' + doc['label'] + '" doc=\'' + JSON.stringify(document) + '\' class="tt-suggestion"></app-suggestion>')(this.scope_);
```

Comment:
I'm nearly there. Now I have access to the document in the directive's
scope, passed through the parent scope, but the compilation doesn't go well
because none of the values are evaluated.

      angular.extend(scope, scope['doc'])
     // here I have all of the doc properties in the scope needed for the
template
      element.html(template(scope['doc']));
      $compile(element.contents())(scope);

And none of the angular bindings show up. Any ideas?

On Wed, Apr 27, 2016 at 2:08 PM, Tobias Sauerwein <notifications@github.com>
wrote:

> In c2corg_ui/static/js/search.js
> <https://github.com/c2corg/v6_ui/pull/311#discussion_r61245941>:
>
> >        suggestion: function(doc) {
> > -        return '<p>' + doc.label + '</p>';
> > -      }
> > -      // Note: The templates for `notFound` and `pending` are not set, because
> > -      // they would be shown for each dataset.
> > +        var document = {
> > +          'query' : doc['_query'],
> > +          'elevation' : doc['elevation'] || doc['elevation_max'],
> > +          'waypoint_type' : doc['waypoint_type'],
> > +          'activities' : doc['activities'],
> > +          'date_end' : doc['date_end'],
> > +          'doctype' : doc['documentType']
> > +        };
> > +        return this.compile_('<app-suggestion label="' + doc['label'] + '" doc=\'' + JSON.stringify(document) + '\' class="tt-suggestion"></app-suggestion>')(this.scope_);
>
> You have a scope with this.scope_. Either you use this scope or create a
> new one with $new
> <https://docs.angularjs.org/api/ng/type/%24rootScope.Scope#%24new>. Then
> set your values to this scope and access them again from your directive.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/c2corg/v6_ui/pull/311/files/03c3a4bc5860d95e944bfd84e4fbea820a2007ba#r61245941>
>



-- 

*Adam Gina*JavaScript Developer
adam.gina@camptocamp.com

*Camptocamp SA*EPFL Innovation Park
Bâtiment A
1015 Lausanne

Tel + 41 21 619 10 10
http://www.camptocamp.com